src/core/vectortile/qgsvectortilewriter.cpp

```diff
@@ -99,35 +110,62 @@ bool QgsVectorTileWriter::writeTiles( QgsFeedback *feedback )
     }
 
     // required metadata
-    mbtiles->setMetadataValue( "name", "???" );  // TODO: custom name?
     mbtiles->setMetadataValue( "format", "pbf" );
+    mbtiles->setMetadataValue( "json", mbtilesJsonSchema() );
 
-    // optional metadata
-    mbtiles->setMetadataValue( "bounds", "-180.0,-85,180,85" );
-    mbtiles->setMetadataValue( "minzoom", QString::number( mMinZoom ) );
-    mbtiles->setMetadataValue( "maxzoom", QString::number( mMaxZoom ) );
-    // TODO: "center"? initial view with [lon,lat,zoom]
+    // metadata specified by the client
+    const QStringList metaKeys = mMetadata.keys();
+    for ( const QString &key : metaKeys )
+    {
+      mbtiles->setMetadataValue( key, mMetadata[key].toString() );
+    }
 
-    // required metadata for vector tiles: "json" with schema of layers
-    mbtiles->setMetadataValue( "json", mbtilesJsonSchema() );
+    // default metadata that we always write (if not written by the client)
+    if ( !mMetadata.contains( "name" ) )
+      mbtiles->setMetadataValue( "name",  "unnamed" );  // required by the spec
+    if ( !mMetadata.contains( "minzoom" ) )
+      mbtiles->setMetadataValue( "minzoom", QString::number( mMinZoom ) );
+    if ( !mMetadata.contains( "maxzoom" ) )
+      mbtiles->setMetadataValue( "maxzoom", QString::number( mMaxZoom ) );
+    if ( !mMetadata.contains( "bounds" ) )
+    {
+      QString boundsStr = "-180,-85,180,85";  // fallback value - whole world except for poles
+      try
+      {
+        QgsCoordinateTransform ct( QgsCoordinateReferenceSystem( "EPSG:3857" ), QgsCoordinateReferenceSystem( "EPSG:4326" ), mTransformContext );
+        QgsRectangle wgsExtent = ct.transform( outputExtent );
+        boundsStr = QString( "%1,%2,%3,%4" )
+                    .arg( wgsExtent.xMinimum() ).arg( wgsExtent.yMinimum() )
+                    .arg( wgsExtent.xMaximum() ).arg( wgsExtent.yMaximum() );
+      }
+      catch ( const QgsCsException & )
+      {
+      }
+      mbtiles->setMetadataValue( "bounds", boundsStr );
```

Comment:
Should this be moved up to sit inside the try block? (Ie if transformation fails then don't set the metadata key)

Reply:
That's probably better than adding default bounds. Thanks for the suggestion, added in 745a537